hack/lib/mesh.bash

```diff
@@ -60,3 +34,70 @@ function undeploy_servicemesh_operators {
   oc delete subscriptions.operators.coreos.com -n openshift-operators servicemeshoperator kiali-ossm jaeger-product --ignore-not-found
 }
 
+function deploy_servicemeshcontrolplane {
+  logger.info "Installing ServiceMeshControlPlane in namespace istio-system"
+
+  oc get ns istio-system || oc create namespace istio-system
+
+  # Make sure servicemeshcontrolplanes.maistra.io is available.
+  timeout 120 "[[ \$(oc get crd servicemeshcontrolplanes.maistra.io --no-headers | wc -l) != 1 ]]" || return 1
+  oc wait --for=condition=Established crd servicemeshcontrolplanes.maistra.io
+
+  # creating smcp often fails due to webhook error
+  timeout 120 "[[ \$(oc apply -f ${resources_dir}/smcp.yaml | oc get smcp -n istio-system basic --no-headers | wc -l) != 1 ]]" || return 1
+  oc wait --timeout=180s --for=condition=Ready smcp -n istio-system basic || oc get smcp -n istio-system basic -o yaml
+}
+
+function undeploy_servicemeshcontrolplane {
+  logger.info "Deleting ServiceMeshControlPlane"
+  oc delete smcp -n openshift-operators basic --ignore-not-found || return $?
+}
+
+function deploy_gateways {
+  oc apply -f "${resources_dir}"/smmr.yaml || return $?
+
+  # Generate wildcard certs with cluster's subdomain.
+
+  local out_dir
+  out_dir="$(mktemp -d /tmp/certs-XXX)"
+
+  oc extract -n openshift-apiserver configmap/config --to="${out_dir}"
+  subdomain=$(yq read "${out_dir}"/config.yaml "routingConfig.subdomain")
```

Comment:
Should this use the `spec.domain` field from `oc get config cluster` instead? You should be able to get away with a `--ojsonpath=".spec.domain"` or thelike I think. That's what our operator is doing as well AFAIK.

Reply:
Sure! updated.